src/openforms/formio/registry.py

```diff
@@ -78,6 +80,26 @@ def mutate_config_dynamically(
     def localize(self, component: ComponentT, language_code: str, enabled: bool):
         pass  # noop by default, specific component types can extend the base behaviour
 
+    def build_serializer_field(self, component: ComponentT) -> serializers.Field:
+        # the default implementation is a compatibility shim while we transition to
+        # the new backend validation mechanism.
+        warnings.warn(
+            "Relying on the default/implicit JSONField for component type "
+            f"{component['type']} is deprecated. Instead, define the "
+            "'build_serializer_field' method on the specific component plugin.",
+            DeprecationWarning,
+        )
```

Comment:
```suggestion
    @typing_extensions.deprecated("msg")
    def build_serializer_field(self, component: ComponentT) -> serializers.Field:
        # the default implementation is a compatibility shim while we transition to
        # the new backend validation mechanism.
```

Identical runtime behavior + typing support :)

Reply:
but breaks my <kbd>CTRL</kbd> + <kbd>F</kb> on `DeprecationWarning` to track down stuff to remove in future releases :/

Reply:
> but breaks my CTRL + F on `DeprecationWarning` to track down stuff to remove in future releases :/

`@deprecated("", category=...)` could do it, but as you said not having access to `component` kills it anyway